infoview/suggestions.tsx

```diff
@@ -0,0 +1,148 @@
+    React.useEffect(() => {
+        const interval = setInterval(() => setCount((count + 1) % 4), 500);
+        return () => clearInterval(interval);
+    }, [count]);
+    return <span>{'.'.repeat(count)}</span>;
+}
+
+function Suggestions(props: {
+    goalState: string, reqId: number, widget: WidgetIdentifier
+}) {
+    const [status, setStatus] = React.useState<SuggestionsInfoStatus>('loading');
+    const [errorMsg, setErrorMsg] = React.useState<string>();
+    const [suggestionsResults, setSuggestionsResults] = React.useState<SuggestionsModelResult>();
+
+    // get Suggestions results - set Suggestions results
+    useEvent(
+        SuggestionsEvent,
+        (suggestions: SuggestionsInt) => {
+            if (suggestions.results && props.reqId === suggestions.reqId) {
+                if (suggestions.results.error) {
+                    setErrorMsg(suggestions.results.error);
+                    setStatus('error');
+                } else {
+                    setSuggestionsResults(suggestions.results);
+                    setStatus('done');
+                }
+            }
+        },
+        [props.reqId]
+    );
+
+    React.useEffect(() => {
+        setStatus('loading');
+        setErrorMsg(undefined); // reset error msg
+    }, [props.reqId]);
+
+    return (
+        <div>
+            <p>
+                <span className={'easeTransition ml2 b-ns ' + statusColTable[status]}>Suggestions</span>
+            </p>
+            <div>
+                {['loading', 'error'].includes(status) ?
+                    <p className={'easeTransition ml3 ' + statusColTable[status]}>
+                        {status === 'loading' ? 'Querying Suggestions' : 'Something went wrong :('}
+                        {status === 'loading'? <Waiter />: null}
+                    </p>
+                : null}
+                {errorMsg ? <i className={'easeTransition ml3 ' + statusColTable[status]}>{errorMsg}</i> : null}
+                {(['done', 'checking'].includes(status) && suggestionsResults && !suggestionsResults.error && suggestionsResults.tactic_infos) ?
+                    <div>{Object.values(suggestionsResults.tactic_infos).map(
+                        tactic_info => <div><SingleTacticInfo key={tactic_info.tactic} tacticInfo={tactic_info} widget={props.widget}/><br/></div>
+                    )}</div>
+                    : null
+                }
+            </div>
+        </div>
+    );
+}
+
+export function Suggestor(props: SuggestionsProps): JSX.Element {
+    const [suggReqId, setSuggReqId] = React.useState(0);
+    const [doSuggest, setDoSuggest] = React.useState(false);
+    const [errorMsg, setErrorMsg] = React.useState<string>();
+    const [prefix, setPrefix] = React.useState('');
+
+    useEvent(
+        SuggestionsErrorEvent,
+        (s) => {
+            setErrorMsg(s.error);
+        }, []
+    );
+
+    React.useEffect(() => {
+        setDoSuggest(true);
+    }, [prefix, props.goalState, props.widget])
```

Comment:
```suggestion
    }, [prefix, props.goalState])
```

This is the reason why it tries to update the suggestion on every cursor move.

React compares the dependency list using `Object.is`, i.e., reference equality.  The `props.widget` field is an object that is fetched every time you move the cursor, so it will be a different reference every time.

We don't need (or read) widgets anywhere, so I think you can just remove all occurrences of `widget`.